### ISSUE TITLE
Python 3.10: ModuleNotFoundError: No module named 'distutils.command.bdist_wininst'

### ISSUE THREAD

**Issue by hroncok** [pypa/setuptools#2558]
Hello. With Python 3.10.0a5+ we have this problem when running setuptools tests in Fedora:

```
==================================== ERRORS ====================================
_____________ ERROR collecting setuptools/command/bdist_wininst.py _____________
setuptools/command/bdist_wininst.py:1: in <module>
    import distutils.command.bdist_wininst as orig
E   ModuleNotFoundError: No module named 'distutils.command.bdist_wininst'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 2 skipped, 1 error in 1.41s ==========================
```

> The bdist_wininst command deprecated in Python 3.8 has been removed. The bdist_wheel command is now recommended to distribute binary packages on Windows. (Contributed by @vstinner in [bpo-42802](https://bugs.python.org/issue42802).)

https://docs.python.org/3.10/whatsnew/3.10.html#distutils

**Comment by hroncok** [comment]
I suppose the import could be try-excepted and the file might do nothing if nothing is there.

Alternatively, the command could be removed from next setuptools version altogether.

**Comment by hroncok** [comment]
I've tried

```
rm setuptools/command/bdist_wininst.py setuptools/tests/test_bdist_deprecations.py
```

But I've been bit by https://github.com/pytest-dev/pytest/pull/8227

**Comment by hroncok** [comment]
This will be tricky.

The distutils deprecation breaks a lot of tests and integration tests try to use old versions of pip that are not compatible with Python 3.10 due to collections.abc aliases removed from collections.

**Comment by hroncok** [comment]
I'Ve drafted https://github.com/pypa/setuptools/pull/2566 to solve my most immediate issues. That PR allows use to continue testing with Python 3.10.0a5 until a permanent solution is discussed.

**Comment by vstinner** [comment]
Oh, I expected setuptools to be compatible with bdist_wininst removal, I prepared the removal in setuptools with:
* commit https://github.com/pypa/setuptools/commit/5d17586a56077dfa3109a5861cf0ff579095a42e
* issue https://github.com/pypa/setuptools/pull/1986

But it seems like I missed this issue.

**Comment by jaraco** [comment]
> integration tests try to use old versions of pip that are not compatible with Python 3.10 due to collections.abc aliases removed from collections

This seems like a separate issue that can be worked around by skipping these tests on Python 3.10 for now. Do we know which versions of pip run on Python 3.10?

**Comment by jaraco** [comment]
> This seems like a separate issue that can be worked around by skipping these tests on Python 3.10 for now. Do we know which versions of pip run on Python 3.10?

Looks like pip 19.2 fixed the issue and 19.1.1 is the last version with the defect.
